distutils/command/bdist.py

```diff
@@ -26,9 +36,10 @@ def show_formats():
 
 class ListCompat(dict[str, tuple[str, str]]):
     # adapter to allow for Setuptools compatibility in format_commands
-    def append(self, item):
+    @deprecated("format_commands is now a dict. append is deprecated.")
+    def append(self, item: object) -> None:
         warnings.warn(
-            """format_commands is now a dict. append is deprecated.""",
+            "format_commands is now a dict. append is deprecated.",
```

Comment:
Thanks for introducing me to this technique. I don't love it, because it's duplicative and adds a lot of boilerplate in the import section, but if this is the best practice, let's keep it.